tests/tracking/test_tracking.py

```diff
@@ -847,3 +848,124 @@ def test_search_runs_multiple_experiments():
     assert len(MlflowClient().search_runs(experiment_ids, "metrics.m_1 > 0", ViewType.ALL)) == 1
     assert len(MlflowClient().search_runs(experiment_ids, "metrics.m_2 = 2", ViewType.ALL)) == 1
     assert len(MlflowClient().search_runs(experiment_ids, "metrics.m_3 < 4", ViewType.ALL)) == 1
+
+
+@pytest.mark.skipif(
+    "MLFLOW_SKINNY" in os.environ,
+    reason="Skinny client does not support the np or pandas dependencies",
+)
+def test_log_table():
+    import pandas as pd
+
+    table_dict = {
+        "inputs": ["What is MLflow?", "What is Databricks?"],
+        "outputs": ["MLflow is ...", "Databricks is ..."],
+        "toxicity": [0.0, 0.0],
+    }
+    artifact_file = "qabot_eval_results.csv"
+    TAG_NAME = "mlflow.loggedArtifacts"
+    run_id = None
+
+    with pytest.raises(
+        MlflowException, match="data must be a pandas.DataFrame or a dictionary"
+    ) as e:
+        with mlflow.start_run() as run:
+            # Log the incorrect data format as a table
+            mlflow.log_table(data="incorrect-data-format", artifact_file=artifact_file)
+    assert e.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
+
+    with mlflow.start_run() as run:
+        # Log the dictionary as a table
+        mlflow.log_table(data=table_dict, artifact_file=artifact_file)
+        run_id = run.info.run_id
+
+    run = mlflow.get_run(run_id)
+    artifact_path = mlflow.artifacts.download_artifacts(run_id=run_id, artifact_path=artifact_file)
+    table_data = pd.read_csv(artifact_path)
+    assert table_data.shape[0] == 2
+    assert table_data.shape[1] == 3
```

Comment:
```suggestion
    pd.testing.assert_frame_equal(table_data, pd.DataFrame(table_dict))
```

We can use `pd.testing.assert_frame_equal` here.